moveit_ros/planning_interface/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -108,10 +108,21 @@ class PlanningComponent
   {
     std::string planner_id;
     std::string planning_pipeline;
-    size_t planning_attempts;
+    int planning_attempts;
     double planning_time;
     double max_velocity_scaling_factor;
     double max_acceleration_scaling_factor;
+
+    void load(const ros::NodeHandle& nh, const std::string& default_planning_pipeline)
+    {
+      std::string ns = "plan_request_params/";
+      nh.param(ns + "planner_id", planner_id, std::string(""));
+      nh.param(ns + "planning_pipeline", planning_pipeline, default_planning_pipeline);
```

Comment:
I think I would leave `default_planning_pipeline` empty by default and simply check for the MoveItCpp configuration.

Reply:
Any opinion on this? I think we should not introduce implicit depencencies in the configurations, so I would leave `planning_pipeline` empty just as `planner_id`

Reply:
I made the default planning pipeline as an empty string because [this line](https://github.com/ros-planning/moveit/blob/master/moveit_ros/planning_interface/moveit_cpp/src/planning_component.cpp#L168) will print an error if it's not specified